components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -209,34 +204,32 @@ std::optional<std::vector<uint8_t>> EthSignTypedDataHelper::EncodeField(
     if (!value.is_list()) {
       return std::nullopt;
     }
-    auto type_split = base::SplitString(type, "[", base::KEEP_WHITESPACE,
-                                        base::SPLIT_WANT_ALL);
+    auto type_split = base::SplitStringPiece(type, "[", base::KEEP_WHITESPACE,
+                                             base::SPLIT_WANT_ALL);
     if (type_split.size() != 2) {
       return std::nullopt;
     }
-    const std::string array_type = type_split[0];
+    auto array_type = type_split[0];
```

Comment:
```suggestion
    const auto array_type = type_split[0];
```
just a nit for it to be `const std::string_view`